org.eclipse.dawnsci.hdf5.model/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModel.java

```diff
@@ -150,27 +172,42 @@ private Object getScalar(String path) {
 			}
 
 			// scalar has not been loaded
-			try (IHierarchicalDataFile reader = getReader.getReader()) {
-				HObject data = (HObject)reader.getData(path);
-				if (data == null) {
+			try {
+				NexusFile reader = null;
+				// test if file exist
+				try {
+					reader = getReader.getReader();
+				} catch (NexusException ne) {
+					if (ne.getMessage().startsWith("File not found"))
+						return null;
+				}
+				if (!reader.isPathValid(path)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "reader" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)